### PR TITLE
Warn the user when they specify a config file / folder that doesn't add any configuration options

### PR DESCRIPTION
**Pull request by cbednarski**

### CODE REVIEW

File: command/agent/command.go [hunk 1]
```diff
@@ -135,6 +136,12 @@ func (c *Command) readConfig() *Config {
 			return nil
 		}
 
+		// The user asked us to load some config here but we didn't find any,
+		// so we'll complain but continue.
+		if current == nil || reflect.DeepEqual(current, &Config{}) {
```
Comment (diptanu) [inline command/agent/command.go]: I think in the case where we couldn't find the configuration file, printing that message distinctly would be better.

Comment (cbednarski) [inline command/agent/command.go]: That case will come back as an error. The specific case this handles is when the user specifies a directory and that directory does not contain any `.hcl` or `.json` files. In that case the LoadConfigDir call returns `&Config{}` and the command fails silently.

If we prefer we can change that case to an error.

Comment (dadgar) [inline command/agent/command.go]: Hm, if they specify a directory and nothing is found, the client/server probably won't load correctly anyways? Maybe we just make it an error

Comment (cbednarski) [inline command/agent/command.go]: Cool I'll change this to an error.

**Comment by cbednarski** [comment]
This should be noted in the changelog under

```
BACKWARDS INCOMPATIBILITIES

- core: `-config` with a directory argument now produces an error if no configuration is loaded from that directory
```

File: command/agent/command.go (hunk 1 above)
Comment (cbednarski) [inline command/agent/command.go]: A consequence of changing this to an error is we cannot support delian-style `service.d` configuration directories.

In other words, you have to rewrite your service file to change configs. I think this makes more sense as a warning which is less stringent about this type of thing.

**Pull request closed without merging**